Development

jafar-sweity opened the pull request.

jafar-sweity committed:
Implement the simplest Add function

jafar-sweity committed:
Add xUnit test project for StringCalculatorKata

jafar-sweity committed:
Implement: Add returns 0 when input is empty & Add_EmptyString_ReturnsZero test function : Add('') should return 0

jafar-sweity committed:
Implement: Add returns single number for input & Add_SingleNumber_ReturnsThatNumber test function

jafar-sweity committed:
Implement: Add returns sum of two comma-separated numbers & Add_TwoNumbersSeparatedByComma_ReturnsSum test function

jafar-sweity committed:
Implement: Add handles unknown number of comma-separated values & Add_MultipleNumbersSeparatedByComma_ReturnsSum test functio

jafar-sweity committed:
Implement: Add supports newline as delimiter & Add_NumbersWithNewLineDelimiter_ReturnsSum test function

jafar-sweity committed:
Implement: Add supports custom delimiter via '//[delimiter]\\n' & Add_CustomDelimiter_ReturnsSum test function

jafar-sweity committed:
Implement: Add throws exception on negative numbers & Add_NegativeNumbers_ThrowsExceptionWithAllNegatives test

jafar-sweity committed:
Implement: Ignore numbers > 1000 & Add_NumbersGreaterThan1000_IgnoredInSum test

oalataiwi commented on StringCalculatorKata.Test/StringCalculatorTest.cs:
```diff
@@ -0,0 +1,66 @@
+        }
+
+        [Fact]
+        public void Add_SingleNumber_ReturnsThatNumber()
```
use Theory plz

oalataiwi commented on StringCalculatorKata.Test/StringCalculatorTest.cs:
```diff
@@ -0,0 +1,66 @@
+            var calculator = new StringCalculator();
+            var ex = Assert.Throws<Exception>(() => calculator.Add("1,-2,3,-4"));
+
+            Assert.Equal("Negative numbers not allowed: -2, -4", ex.Message);
```
put msg in separate file

oalataiwi commented on StringCalculatorKata/Services/StringCalculator.cs:
```diff
@@ -1,12 +1,39 @@
+            }
+
+            var numberList = numbers.Split(delimiters, StringSplitOptions.None);
+            var parsedNumbers = numberList.Select(int.Parse).ToList();
```
use try parse

jafar-sweity committed:
refactor(tests): merge individual Add tests into single theory with descriptive naming

- Consolidated multiple Add method tests into a single Theory using InlineData
- Renamed test method to follow should naming convention for clarity

jafar-sweity committed:
refactor : use tryparse to avoid exceptions

jafar-sweity committed:
refactor : use centralized error message
